test/unit/org/apache/cassandra/index/IndexStatusManagerTest.java

```diff
@@ -0,0 +1,393 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import org.apache.cassandra.exceptions.ReadFailureException;
+import org.apache.cassandra.gms.VersionedValue;
+import org.apache.cassandra.locator.*;
+import org.apache.cassandra.utils.JsonUtils;
+import org.junit.Test;
+
+import org.apache.cassandra.db.ConsistencyLevel;
+import org.apache.cassandra.db.Keyspace;
+import org.apache.cassandra.schema.IndexMetadata;
+import org.mockito.Mockito;
+
+import static org.apache.cassandra.locator.ReplicaUtils.full;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.Assert.assertArrayEquals;
+
+public class IndexStatusManagerTest
+{
+    static class Testcase
+    {
+        String keyspace;
+        int minRequiredNodes;
+        AbstractReplicationStrategy replicationStrategy;
+        Map<InetAddressAndPort, Map<String, Index.Status>> indexStatus;
+        EndpointsForRange expected;
+
+        Testcase(Builder builder)
+        {
+            keyspace = builder.keyspace;
+            minRequiredNodes = builder.minRequiredNodes;
+            replicationStrategy = builder.replicationStrategy;
+            indexStatus = builder.indexStatus;
+            expected = builder.expected;
+        }
+
+        static class Builder
+        {
+            String keyspace;
+            int minRequiredNodes;
+            AbstractReplicationStrategy replicationStrategy;
+            Map<InetAddressAndPort, Map<String, Index.Status>> indexStatus;
+            EndpointsForRange expected;
+
+            Builder keyspace(String kp)
```

Comment:
```suggestion
            Builder keyspace(String ks)
```
super nit: We usually use "ks" as an abbreviation for keyspace, but no important